Sources/AWSLambdaRuntimeCore/DetachedTasks.swift

```diff
@@ -0,0 +1,169 @@
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import Foundation
+import NIOConcurrencyHelpers
+import NIOCore
+import Logging
+
+/// A container that allows tasks to finish after a synchronous invocation
+/// has produced its response.
+public final class DetachedTasksContainer {
+
+    struct Context {
+        let eventLoop: EventLoop
+        let logger: Logger
+    }
+    
+    private var context: Context
+    private var storage: Storage
+
+    init(context: Context) {
+        self.storage = Storage()
+        self.context = context
+    }
+
+    /// Adds a detached task that runs on the given event loop.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the task.
+    ///   - task: The task to execute. It receives an `EventLoop` and returns an `EventLoopFuture<Void>`.
+    /// - Returns: A `RegistrationKey` for the registered task.
+    @discardableResult
+    public func detached(name: String, task: @escaping (EventLoop) -> EventLoopFuture<Void>) -> RegistrationKey {
+        let key = RegistrationKey()
+        let task = task(context.eventLoop).always { _ in
+            self.storage.remove(key)
+        }
+        self.storage.add(key: key, name: name, task: task)
+        return key
+    }
+    
+    /// Adds a detached async task.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the task.
+    ///   - task: The async task to execute.
+    /// - Returns: A `RegistrationKey` for the registered task.
+    @discardableResult
+    public func detached(name: String, task: @Sendable @escaping () async throws -> Void) -> RegistrationKey {
+        let key = RegistrationKey()
+        let promise = context.eventLoop.makePromise(of: Void.self)
+        promise.completeWithTask(task)
+        let task = promise.futureResult.always { result in
+            switch result {
+            case .success:
+                break
+            case .failure(let failure):
+                self.context.logger.warning(
+                    "Execution of detached task failed with error.",
+                    metadata: [
+                        "taskName": "\(name)",
+                        "error": "\(failure)"
+                    ]
+                )
+            }
+            self.storage.remove(key)
+        }
+        self.storage.add(key: key, name: name, task: task)
+        return key
+    }
+
+    /// Informs the runtime that the specified task should not be awaited anymore.
+    ///
+    /// - Warning: This method does not actually stop the execution of the
+    ///   detached task, it only prevents the runtime from waiting for it before
+    ///   `/next` is invoked.
+    ///
+    /// - Parameter key: The `RegistrationKey` of the task to cancel.
+    public func unsafeCancel(_ key: RegistrationKey) {
```

Comment:
We are not using structured concurrency, for this reason we should not try to add cancel support. Cancel support will be added when we move to a fully async/structured concurrency api.